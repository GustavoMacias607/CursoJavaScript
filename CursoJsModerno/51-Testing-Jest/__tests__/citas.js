import Citas from '../js/classes/Citas'


describe('Probar la clase de citas', () => {

    const citas = new Citas()
    const id = Date.now()
    test('agregar una nueva cita', () => {
        const citaObj = {
            mascota: 'Canelo',
            propietario: 'Gus',
            telefono: '12345',
            fecha: '10-12-2023',
            hora: '10:30',
            sintomas: 'solo duerme'
        }

        citaObj.id = id
        citas.agregarCita(citaObj)

        //prueba
        expect(citas).toMatchSnapshot()



    })
    test('actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Candela',
            propietario: 'Gus',
            telefono: '12345',
            fecha: '10-12-2023',
            hora: '10:30',
            sintomas: 'solo duerme'
        }
        citas.editarCita(citaActualizada)

        expect(citas).toMatchSnapshot()
    })

    test('Elimiar cita', () => {
        citas.eliminarCita(id)
        expect(citas).toMatchSnapshot()
    })
})