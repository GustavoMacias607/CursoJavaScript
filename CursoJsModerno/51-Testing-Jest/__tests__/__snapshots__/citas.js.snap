// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Elimiar cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "10:30",
      "id": 1689983592092,
      "mascota": "Candela",
      "propietario": "Gus",
      "sintomas": "solo duerme",
      "telefono": "12345",
    },
  ],
}
`;

exports[`Probar la clase de citas agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "10:30",
      "id": 1689983592092,
      "mascota": "Canelo",
      "propietario": "Gus",
      "sintomas": "solo duerme",
      "telefono": "12345",
    },
  ],
}
`;
